notifications/views.py

```diff
@@ -2,13 +2,17 @@
 from django.forms import ValidationError
 from django_filters.rest_framework import DjangoFilterBackend
 from django.contrib.auth import get_user_model
+from django.shortcuts import get_object_or_404
+
+from rest_framework import generics, filters, status
 
 from rest_framework import generics, filters, status
 from rest_framework.generics import get_object_or_404
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.exceptions import PermissionDenied, NotFound
 from rest_framework.exceptions import NotFound
```

Comment:
There are several duplicate imports, such as IsAuthenticated, get_user_model, and NotFound.